### ISSUE TITLE
`@adt` macro increases the time to second using

### ISSUE THREAD

**Issue by GiggleLiu** [Jugsaw/Jugsaw#77]
```julia
julia> @time using JugsawIR
[ Info: Precompiling JugsawIR [6340f1c1-b722-4800-8081-8412e0693b45]
 13.280598 seconds (1.65 M allocations: 106.299 MiB, 1.22% gc time, 12.17% compilation time: 92% of which was recompilation)

julia> @time using JugsawIR
  2.536745 seconds (1.90 M allocations: 132.819 MiB, 1.66% gc time, 100.00% compilation time: 100% of which was recompilation)
```

@Roger-luo If I remote the `@adt`, the time to second using reduced to ~0. The time to first using is also increased a lot by `@adt`.

Do you have any idea to fix this issue?

**Comment by Roger-luo** [comment]
this seems due to recompilation - somehow it recompiles everything when you load this module, I don't understand why however, what `@adt` does is just overload a bunch of methods for you.

**Comment by GiggleLiu** [comment]
Yeah, it is wield. I will probably implement it manually later.

**Comment by GiggleLiu** [comment]
I found the problem, overloading `getproperty(::Type{XXX}, symbol)` causes recompilation of the whole package. I have to construct different types explicitly.

**Comment by GiggleLiu** [comment]
Fixed in #82
